x/stakeibc/types/message_toggle_trade_controller.go

```diff
@@ -0,0 +1,64 @@
+package types
+
+import (
+	"errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	errorsmod "cosmossdk.io/errors"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/Stride-Labs/stride/v19/utils"
+)
+
+const TypeMsgToggleTradeController = "toggle_trade_controller"
+
+var _ sdk.Msg = &MsgToggleTradeController{}
+
+func NewMsgToggleTradeController(creator, chainId string, permissionChange AuthzPermissionChange, address string) *MsgToggleTradeController {
+	return &MsgToggleTradeController{
+		Creator:          creator,
+		ChainId:          chainId,
+		PermissionChange: permissionChange,
+		Address:          address,
+	}
+}
+
+func (msg *MsgToggleTradeController) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgToggleTradeController) Type() string {
+	return TypeMsgToggleTradeController
+}
+
+func (msg *MsgToggleTradeController) GetSigners() []sdk.AccAddress {
+	creator, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{creator}
+}
+
+func (msg *MsgToggleTradeController) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgToggleTradeController) ValidateBasic() error {
```

Comment:
should we add another check on the enum here or are the checks in `tx_toggle_trade_controller` enough?

Reply:
good call, added